src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -372,147 +372,277 @@ module Base = struct
     let%bind then_ = then_ and else_ = else_ in
     if_ b ~then_ ~else_
 
-  (* spec for
-     [apply_tagged_transaction root (tag, { sender; signature; payload }]):
-     - if tag = Normal:
-        - check that [signature] is a signature by [sender] of payload
-        - return:
-          - merkle tree [root'] where the sender balance is decremented by
-            [payload.amount] and the receiver balance is incremented by [payload.amount].
-          - fee excess = +fee.
-          -if coinbase, then push it to the stack [pending_coinbase_stack_before]
-
-     - if tag = Fee_transfer
-        - return:
-          - merkle tree [root'] where the sender balance is incremented by
-            fee and the receiver balance is incremented by amount
-          - fee excess = -(amount + fee)
-
-  *)
-  (* Nonce should only be incremented if it is a "Normal" transaction. *)
   let%snarkydef apply_tagged_transaction (type shifted)
       (shifted : (module Inner_curve.Checked.Shifted.S with type t = shifted))
       root pending_coinbase_stack_before pending_coinbase_after
       state_body_hash_opt
-      ({sender; signature; payload} : Transaction_union.var) =
+      ({signer; signature; payload} : Transaction_union.var) =
+    let%bind is_user_command =
```

Comment:
do we have unit tests (preferably quickcheck tests) that cover these changes? If not, how hard would it be to add?

Reply:
Pushed unit tests for the transaction snark.